### DISCUSSION TITLE
Workflow for plotting Maps in Valhalla

### DISCUSSION THREAD

**Discussion by ghost** [valhalla/valhalla#3193] [Q&A]
(As this might be a very basic question, please delete it, if it's not suitable)

I'm currently working with Valhalla (docker build, see [here](https://gis-ops.com/valhalla-how-to-run-with-docker-on-ubuntu/)), however I don't understand how I can plot my resulting trips.

E.g. If I query using
`curl http://localhost:8002/route --data '{"locations":[{"lat":47.029,"lon":15.4327},{"lat":47.0332,"lon":15.4132}],"costing":"auto","directions_options":{"units":"kilometers"}}' | jq '.'`

I receive the output [valhalla_out.txt](https://github.com/valhalla/valhalla/files/6763687/valhalla_out.txt)

But how do I plot this output in a map/ What kind of workflow would you recommend for plotting multiple trips?

Thank you for your reply!

**Comment by kevinkreiser** [comment]
https://github.com/valhalla/valhalla-docs/issues/203#issuecomment-358845083

The second link is to a deprecated repo but we moved it to this repo. Also I think the general docs mention this https://github.com/valhalla/valhalla/blob/master/docs/api/turn-by-turn/api-reference.md#trip-legs-and-maneuvers ~but actually the link is broken~ I fixed it

**Reply by ghost** [reply]
Thanks for your reply!

Just to sum things up: If I got it right, a recommended workflow for plotting multiple trips (e.g. A->B, C->D, E->F) would be something like:

1. Get routes using `localhost:8002/optimized_route?json={}`
2. Extract Polyline and decode it using `https://github.com/valhalla/valhalla-docs/blob/master/decoding.md`
3. Save all encoded Polylines to a geojson and plot it

Is this what you meant?

**Reply by kevinkreiser** [reply]
im not sure what you mean for multiple trips. `optimized_route` will try to reorder the locations A B C D E F... to be the optimal ordering so that the final route is the shortest (for some value of short). if you want to just do A  to B to C to D etc. then you should call `route` and put the locations in that order. i dont know if you specifically need to do this in geojson but if you just want to visualize the geometry of the route we already have demos that do that: http://valhalla.github.io/demos/routing/simple.html

the code for the demo is also availble: https://github.com/valhalla/demos/blob/gh-pages/routing/simple.html
